File: app/src/main/java/com/supercilex/robotscouter/util/ConnectivityUtils.java
package com.supercilex.robotscouter.util;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

public enum ConnectivityUtils {;
    public static boolean isOffline(Context context) {
        ConnectivityManager connectivityManager =
                (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo activeNetworkInfo = connectivityManager.getActiveNetworkInfo();
        return !(activeNetworkInfo != null && activeNetworkInfo.isConnected());
    }
}

File: app/src/main/java/com/supercilex/robotscouter/util/connectivityUtils.kt
package com.supercilex.robotscouter.util

import android.content.Context
import android.net.ConnectivityManager
import com.supercilex.robotscouter.RobotScouter

private val connectivityManager by lazy {
    RobotScouter.INSTANCE.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
}

fun isOffline() = connectivityManager.activeNetworkInfo?.isConnected == false
